modules/components/org.wso2.carbon.appfactory.resource.mgt/src/main/java/org/wso2/carbon/appfactory/resource/mgt/listeners/TenantCreationDurableSubscriber.java

```diff
@@ -0,0 +1,99 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.carbon.appfactory.resource.mgt.listeners;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.appfactory.common.AppFactoryConstants;
+import org.wso2.carbon.appfactory.common.AppFactoryException;
+import org.wso2.carbon.appfactory.eventing.AppFactoryEventException;
+import org.wso2.carbon.appfactory.eventing.utils.Util;
+
+import javax.jms.*;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.util.Properties;
+
+/**
+ * Durable subscriber for topics  in MB - tenant creation.
+ * TenantCreatioMessageListener is set as the message listener
+ */
+public class TenantCreationDurableSubscriber {
+    private static Log log = LogFactory.getLog(TenantCreationDurableSubscriber.class);
+    private String subscriptionId;
+    private String topicName;
+
+    public TenantCreationDurableSubscriber(String topicName, String subscriptionId)
+            throws AppFactoryException {
+        this.topicName = topicName;
+	    this.subscriptionId = subscriptionId;
+        try {
+            subscribe();
+        } catch (AppFactoryEventException e) {
+	        throw new AppFactoryException("Subscriber activation failed for topic" + topicName + " and subscription " +
+	                                      "id :  " + subscriptionId, e);
+        }
+    }
+
+    /**
+     * Subscribe as a durable subscriber to the topic.
+     *
+     * @throws AppFactoryEventException
+     */
+    public void subscribe() throws AppFactoryEventException {
+        Properties properties = new Properties();
+        properties.put(Context.INITIAL_CONTEXT_FACTORY, AppFactoryConstants.ANDES_ICF);
+        properties.put(AppFactoryConstants.CF_NAME_PREFIX + AppFactoryConstants.CF_NAME, Util.getTCPConnectionURL());
+        properties.put(CarbonConstants.REQUEST_BASE_CONTEXT, "true");
+        properties.put(AppFactoryConstants.TOPIC, topicName);
+        TopicConnectionFactory connFactory;
+        TopicConnection topicConnection;
+        TopicSession topicSession;
+        TopicSubscriber topicSubscriber;
+        InitialContext ctx;
+        try {
+            ctx = new InitialContext(properties);
+            connFactory = (TopicConnectionFactory) ctx.lookup(AppFactoryConstants.CF_NAME);
+            topicConnection = connFactory.createTopicConnection();
+            topicSession = topicConnection.createTopicSession(false, TopicSession.CLIENT_ACKNOWLEDGE);
+            Topic topic;
+            try {
+                topic = (Topic) ctx.lookup(topicName);
+            } catch (NamingException e) {
+                topic = topicSession.createTopic(topicName);

```

Comment:
Why we are not logging the exception here? If it is not needed, add comment stating the reason.